knative/describer.go

```diff
@@ -1,69 +1,57 @@
 package knative
 
 import (
-	"fmt"
 	"k8s.io/apimachinery/pkg/api/errors"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/rest"
+	v1 "knative.dev/client/pkg/serving/v1"
 	"knative.dev/eventing/pkg/apis/eventing/v1beta1"
-	eventingv1client "knative.dev/eventing/pkg/client/clientset/versioned/typed/eventing/v1beta1"
-	servingv1client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1alpha1"
 
 	"github.com/boson-project/faas"
 	"github.com/boson-project/faas/k8s"
 )
 
 type Describer struct {
-	Verbose        bool
-	namespace      string
-	servingClient  *servingv1client.ServingV1alpha1Client
-	eventingClient *eventingv1client.EventingV1beta1Client
-	config         *rest.Config
+	Verbose   bool
+	namespace string
 }
 
 func NewDescriber(namespaceOverride string) (describer *Describer, err error) {
 	describer = &Describer{}
-	config, namespace, err := newClientConfig(namespaceOverride)
+	namespace, err := GetNamespace(namespaceOverride)
 	if err != nil {
 		return
 	}
-	describer.namespace = namespace
 
-	describer.servingClient, err = servingv1client.NewForConfig(config)
-	if err != nil {
-		return
-	}
-	describer.eventingClient, err = eventingv1client.NewForConfig(config)
-	if err != nil {
-		return
-	}
-	describer.config = config
+	describer.namespace = namespace
 	return
 }
 
 // Describe by name. Note that the consuming API uses domain style notation, whereas Kubernetes
 // restricts to label-syntax, which is thus escaped. Therefore as a knative (kube) implementation
 // detal proper full names have to be escaped on the way in and unescaped on the way out. ex:
 // www.example-site.com -> www-example--site-com
-func (describer *Describer) Describe(name string) (description faas.Description, err error) {
-
-	namespace := describer.namespace
-	servingClient := describer.servingClient
-	eventingClient := describer.eventingClient
+func (d *Describer) Describe(name string) (description faas.Description, err error) {
 
 	serviceName, err := k8s.ToK8sAllowedName(name)
 	if err != nil {
 		return
 	}
 
-	service, err := servingClient.Services(namespace).Get(serviceName, metav1.GetOptions{})
+	servingClient, _, err := NewServingClient(d.namespace, d.Verbose)
```

Comment:
This is the canonical constructor that should be used to avoid the dependency on the commands.

Reply:
This is the constructor from `knative/client.go` https://github.com/boson-project/faas/pull/167/files/62c488923ae8ce21b3d6b49d49e80cc27282763d#diff-cc862f411ab104c6c73b4f523a13b5a516a6ac6ceaf8edd7bcc664ec12b33889R20